addons/html_editor/static/src/fields/html_field.js

```diff
@@ -6,27 +6,55 @@ import { registry } from "@web/core/registry";
 import { useBus } from "@web/core/utils/hooks";
 import { useRecordObserver } from "@web/model/relational_model/utils";
 import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { HtmlViewer } from "./html_viewer";
+
+/**
+ * Check whether the current value contains nodes that would break
+ * on insertion inside an existing body.
+ *
+ * @returns {boolean} true if 'this.props.value' contains a node
+ * that can only exist once per document.
+ */
+function computeContainsComplexHTML(value) {
```

Comment:
would it make sense to move this into htmlviewer?

Reply:
Actually, we need it to know if we are in readonly or not. But we can may remove it and alway pass the option sandboxPreview